server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletManagementIterator.java

```diff
@@ -260,11 +260,6 @@ private void computeTabletManagementActions(final TabletMetadata tm,
       reasonsToReturnThisTablet.add(ManagementAction.NEEDS_VOLUME_REPLACEMENT);
     }
 
-    if (!reasonsToReturnThisTablet.isEmpty()) {
```

Comment:
why remove this, why not return early?

Reply:
When a tablet was assigned to a dead tserver, it would not do volume replacement and if it returned early here the TGW code would never remove the dead tserver location.

Reply:
Ok, do we need to modify TGW to not split or compact then NEEDS_RECOVERY is in the action list?

Reply:
Or, maybe, don't evaluate those conditions below is a better option.

Reply:
pushed a change in 0e24090 to not evaluation the split and compaction when certain actions are set